src/main/java/mekanism/common/world/ResizableDiskConfig.java

```diff
@@ -2,30 +2,30 @@
 
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
-import java.util.List;
 import java.util.function.IntSupplier;
 import mekanism.common.config.MekanismConfig;
 import mekanism.common.config.WorldConfig.SaltConfig;
 import net.minecraft.util.valueproviders.IntProvider;
 import net.minecraft.world.level.block.Blocks;
-import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.levelgen.blockpredicates.BlockPredicate;
 import net.minecraft.world.level.levelgen.feature.configurations.FeatureConfiguration;
+import net.minecraft.world.level.levelgen.feature.stateproviders.RuleBasedBlockStateProvider;
 
 public class ResizableDiskConfig implements FeatureConfiguration {
 
     public static final Codec<ResizableDiskConfig> CODEC = RecordCodecBuilder.create(builder -> builder.group(
-          BlockState.CODEC.fieldOf("state").forGetter(config -> config.state)
-    ).apply(builder, state -> new ResizableDiskConfig(state, MekanismConfig.world.salt)));
+            RuleBasedBlockStateProvider.CODEC.fieldOf("state_provider").forGetter(config -> config.stateProvider)
+    ).apply(builder, stateProvider -> new ResizableDiskConfig(stateProvider, MekanismConfig.world.salt)));
 
-    public final BlockState state;
+    public final RuleBasedBlockStateProvider stateProvider;
     public final IntProvider radius;
     public final IntSupplier halfHeight;
```

Comment:
This file changed to match the vanilla config as closely as possible. The only remaining difference is the `IntSupplier` since I am not aware of any clean approach towards making `configured_feature/salt.json` pull in the Mekanism config value for the `radius`.